spec/controller/open_library_controller_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'rails_helper'
+require 'json'
+
+RSpec.describe Api::V1::OpenLibraryController do
+  describe 'GET Correct body response' do
+    before do
+      @stubbed_service = instance_double(OpenLibraryService)
+      book_info = File.read('./spec/mock/book_info.json')
+      allow(@stubbed_service).to receive(:book_info).and_return(JSON.parse(book_info))
+      allow(OpenLibraryService).to receive(:new).and_return(@stubbed_service)
+    end
+    describe 'GET body response' do
+      it 'Correct body response' do
+        book_info = File.read('./spec/mock/book_info.json')
+        book_info_service = OpenLibraryService.new('0385472579').book_info
+        expect(book_info_service).to eq(JSON.parse(book_info))
+      end
+    end
+  end
+end
+RSpec.describe Api::V1::OpenLibraryController do
+  describe 'GET Bad body response' do
+    before do
+      @stubbed_bad_service = instance_double(OpenLibraryService)
+      allow(@stubbed_bad_service).to receive(:book_info).and_return({})
+      allow(OpenLibraryService).to receive(:new).and_return(@stubbed_bad_service)
+    end
+    describe 'body response' do
+      it 'BAD body response' do
+        book_info_service = OpenLibraryService.new('bad_isbn').book_info
+        expect(book_info_service).to eq({})
+      end
+    end
+  end
+end
```

Comment:
this is only for suggestion that how to manage the test files, important separate the describe section and remember use Fictium like 
```ruby
describe PostsController do
  describe action 'GET #index' do # `action` after describe define the method that you make testing
    describe example 'without errors' do # `example` after describe define that is an example of the action
        default_example # You can select an example to be marked as default in your action

        before do
            get :index
        end

        # ... your tests
    end
  end
end
```

is possible to move the mock logic to shared context

```suggestion
RSpec.describe Api::V1::OpenLibraryController do
  describe action 'GET book info' do
    before do
      @stubbed_service = instance_double(OpenLibraryService)
      allow(@stubbed_service).to receive(:book_info).and_return(JSON.parse(book_info))
      allow(OpenLibraryService).to receive(:new).and_return(@stubbed_service)
    end

    describe example 'with valid isbn' do
      default_example

      it { expect(response).to have_http_status(:ok) }

      it 'Correct body response' do
        let(:book_info) { File.read('./spec/mock/book_info.json') }
        book_info_service = OpenLibraryService.new('0385472579').book_info
        expect(book_info_service).to eq(JSON.parse(book_info))
      end
    end

    describe example 'with bad isbn' do

      it { expect(response).to have_http_status(:not_found) }
  
      it 'BAD body response' do
        let(:book_info) { {} }
        book_info_service = OpenLibraryService.new('bad_isbn').book_info
        expect(book_info_service).to eq({})
      end
    end
  end
end
```

Reply:
Also remember to write the action when describing these tests. It should be able to be read as an entire sentence without interruptions

For example: 

it 'returns bad body response'